src/Executor/Executor.php

```diff
@@ -10,6 +10,7 @@
 use GraphQL\Language\AST\NodeKind;
 use GraphQL\Language\AST\OperationDefinitionNode;
 use GraphQL\Language\AST\SelectionSetNode;
+use GraphQL\Promise\PromiseAdapterInterface;
```

Comment:
Please rename interface to `PromiseAdapter`. We don't use `Interface` suffix anywhere else in this code base. Also we should move `Promise` namespace to `Executor\Promise` as it will only be used during Execution step.